run.ps1

```diff
@@ -0,0 +1,34 @@
+# set the base tools directory
+$toolsLocalPath = Join-Path (Get-Location) "Tools"
+$runExe = Join-Path $toolsLocalPath "Microsoft.DotNet.BuildTools.Run\run.exe"
+
+# if the tool doesn't exist then download and run the boot-strapper script
+if ((Test-Path $runExe) -eq 0)
```

Comment:
This is adding another layer of semaphore checking which will block updating tools if .toolversion updates. I think we need to make some more changes to this script to always call bootstrap.ps1, and presumable cache it in some way for subsequent runs.